sklearn/model_selection/_split.py

```diff
@@ -1472,6 +1472,11 @@ def _iter_indices(self, X, y, groups=None):
         y = check_array(y, ensure_2d=False, dtype=None)
         n_train, n_test = _validate_shuffle_split(n_samples, self.test_size,
                                                   self.train_size)
+
+        if y.ndim == 2:
+            # for multi-label y, map each distinct row to a unique identifier:
+            y = np.array([hash(tuple(row)) for row in y])
```

Comment:
I had meant to point out that python's hash is far from a uniform/unique hash, but I don't think that's a big concern here